### PR TITLE
Fix for permission errors in cachedir and pkidir when running minion as non-root

### PR DESCRIPTION
**Pull request by robin-wittler**
Greetings,

i've run into this when trying to run salt-minion (v2014.7.1) as a non-root.
Problem was that get_proc_dir creates cachedir/proc as user root and after
changing the user (after the double fork etc. pp)  this user can not read cachedir/proc.

I made this non-intrusive (hopefully ... haha), that means you can call get_proc_dir with
no change of the previous default behavior at all.

I also extended salt.utils with get_uid and get_gid convinient functions
and use salt.utils.get_uid in the Minion and ProxyMinion Classes (when they
call get_proc_dir in **init**).

I do not know if i should open a issue for this, for me it was more important to fix this.
There is another minion permission issue in combination with those options:

verify_master_pubkey_sign: True
always_verify_signature: True

The minion_master.pub in the minion pki dir gets created as root and then, again after the fork and user setting, the minion is no longer able to read/write to minion_master.pub. I will also have a fix for it, but i
just wanne know, if i can go ahead with the style and so on, or do i need something to change?

### CODE REVIEW

File: salt/minion.py [hunk 1]
```diff
@@ -137,15 +137,60 @@ def resolve_dns(opts):
...
+        d_stat = os.stat(fn_)
+    else:
+        d_stat = os.stat(fn_)
+        # if mode is not an empty dict then we have an explicit
+        # dir mode. So lets check if mode needs to be changed.
+        if mode:
+            if d_stat.st_mode | mode['mode'] != d_stat.st_mode:
+                os.chmod(fn_, d_stat.st_mode | mode['mode'])
+
+    if hasattr(os, 'chmod'):
```
Comment (robin-wittler) [inline salt/minion.py]: Well this line should be changed to 

if hasattr(os, 'chown'):

**Comment by jfindlay** [comment]
Go Go Jenkins!

**Comment by jfindlay** [comment]
Go Go Jenkins!

**Comment by jfindlay** [comment]
@robin-wittler, please rebase on develop as there have been test fixes that have come in after you forked from develop, thanks.

**Pull request closed without merging**